model/src/main/kotlin/DependencyGraphNavigator.kt

```diff
@@ -49,9 +51,13 @@ class DependencyGraphNavigator(
         //       they manage, for example that "GradleInspector" manages "Gradle" projects.
         val managers = graphs.keys.filter { it.startsWith(project.id.type) }
 
-        val manager = requireNotNull(managers.singleOrNull()) {
-            "All of the $managers managers are able to manage '${project.id.type}' projects. Please enable only one " +
-                "of them."
+        val manager = managers.first()
```

Comment:
Commit: Why have you chosen that ORT should continue rather than logging the error and re-throwing the exception? For the code in `Pub` I blind guess that the problem this may cause is limited. However, here in the dependency graph, I wonder what kind of mis-behavior can be cause by this tolerance. That'd depends on the usages of this function. Can you give a hint why it is ok?

Reply:
> However, here in the dependency graph, I wonder what kind of mis-behavior can be cause by this tolerance.

I'm wondering the same, and I agree that it's an error that should never happen, but [apparently it does](https://oss-review-toolkit.slack.com/archives/C9NNJ54B1/p1725966025817149). I order to not block our users while we're investigating the root cause, I decided to just make this a "soft" error log for now, as it's also a bit difficult to file an issue from the `DependencyGraphNavigator` without bigger refactoring.

Once the root cause is found and fixed, we should consider making this a "hard" failure again.

Reply:
Hm, I'm struggling a bit with confidence of approving it. However, back to the topic
- From the slack discussion, am I right that the error message does not originate from the code location in Pub
  but from this dep graph navigator?
- As root cause is unclear, and future investigation will be needed, I propose to create a GitHub issue for that bug, so that we can link from the commit / PR, so that it is easy to find later on.

Reply:
> am I right that the error message does not originate from the code location in Pub
but from this dep graph navigator?

Yes. However, that was only found out as part of the discussion.

> As root cause is unclear

I've found the root cause meanwhile, see https://github.com/oss-review-toolkit/ort/pull/9234.

Reply:
I took a look, but I haven't understood it yet. Did the unexpected type value come from a project which does not use the graph representation?

However, intention is to defer review of this one until that other PR is ready?

Reply:
> Did the unexpected type value come from a project which does not use the graph representation?

No. The problem is that the keys in the graph map should be the name of the package managers (like "GradleInspector"), and not the name of the type of projects the package manager manages (like "Gradle"). The logic in `resolvePackageManagerDependencies()` breaks that convention.

> However, intention is to defer review of this one until that other PR is ready?

It depends on how quickly I'm able to come up with the real fix. I believe this PR is still valuable for people looking for a work-around of the current issue. So I'll not close the PR yet, but there is also no rush to get it merged.

Reply:
Thanks for explaining!